packages/astro/src/runtime/server/endpoint.ts

```diff
@@ -23,38 +11,34 @@ export async function renderEndpoint(
 ) {
 	const { request, url } = context;
 
-	const chosenMethod = request.method?.toUpperCase();
-	const handler = getHandlerFromModule(mod, chosenMethod);
-	if (!ssr && ssr === false && chosenMethod && chosenMethod !== 'GET') {
+	const method = request.method.toUpperCase();
+	// use the exact match on `method`, fallback to ALL
+	const handler = mod[method] ?? mod['ALL'];
+	if (!ssr && ssr === false && method !== 'GET') {
 		logger.warn(
-			null,
+			"router",
 			`${url.pathname} ${bold(
-				chosenMethod
+				method
 			)} requests are not available for a static site. Update your config to \`output: 'server'\` or \`output: 'hybrid'\` to enable.`
 		);
 	}
-	if (!handler || typeof handler !== 'function') {
+	if (typeof handler !== 'function') {
+		logger.warn(
+			'router',
+			`No API Route handler exists for the method "${method}" for the route ${url.pathname}.\n` +
+			`Found handlers: ${Object.keys(mod).map(exp => JSON.stringify(exp)).join(', ')}\n` +
+			('all' in mod ? `One of the exported handlers is "all" (lowercase), did you mean to export 'ALL'?\n` : '')
```

Comment:
We **removed** the support lower-case functions in v4 (they were deprecated in v3), so we shouldn't show a warning anymore, otherwise it sounds like we still support it.

Reply:
This message would only be shown when `pages/x.ts` existed but `/x` failed to match a handler. I think at that point, it's safe to say the user could use a helpful nudge.

Reply:
To be more precise, I talk about the `all` check. We don't support that anymore, so we shouldn't write **new** code around that. New developers would see this line and won't understand why it is there.